rustls/tests/api.rs

```diff
@@ -3958,6 +3958,64 @@ fn test_client_sends_helloretryrequest() {
     ));
 }
 
+#[test]
+fn test_server_requests_retry_with_echoed_session_id() {
+    use rustls::internal::msgs::handshake::SessionId;
+    let expected_session_id = SessionId::random::<Ring>().unwrap();
```

Comment:
Oops, fixed this `<Ring>` generic and `SessionId` bit in the follow-up commit instead of this one. Will rebase and adjust shortly.